preprocessing/nextclade/src/loculus_preprocessing/prepro.py

```diff
@@ -69,12 +69,28 @@ def null_per_backend(x: Any) -> bool:
             return False
 
 
+def get_segment(spec: ProcessingSpec, unprocessed: UnprocessedAfterNextclade) -> str | None:
+    """Returns the segment to use based on spec args"""
+    if spec.args and spec.args.get("useFirstSegment", False) and unprocessed.nextcladeMetadata:
+        valid_segments = [key for key, value in unprocessed.nextcladeMetadata.items() if value]
+        if not valid_segments:
+            return None
+        if len(valid_segments) > 1:
```

Comment:
I think this error would actually have to be returned in the processing result as an error (so that the user sees it and contacts the administrator). We have some open todos in the add_nextclade_metadata function for this - could you maybe add a todo here and I will draft a PR that returns proper errors in these cases?

Reply:
Could you check whether the generated error annotation looks ok?